src/Casdoor.Client/CasdoorHttpClient.cs

```diff
@@ -0,0 +1,95 @@
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using Casdoor.Client.Config;
+using Casdoor.Client.Entity;
+using Casdoor.Client.Exception;
+using Casdoor.Client.Util;
+
+namespace Casdoor.Client;
+
+public class CasdoorHttpClient
+{
+    private readonly CasdoorClientOptions _options;
+
+    public CasdoorHttpClient(CasdoorClientOptions? options)
+    {
+        _options = options ?? throw new ArgumentNullException(nameof(options));
+    }
+
+    public async Task<string> DoGetStringAsync(string url)
+    {
+        using var handler = new HttpClientHandler
+        {
+            Credentials = new NetworkCredential(_options.ClientId, _options.ClientSecret)
+        };
+        using var client = new HttpClient(handler);
+        return await client.GetStringAsync(url);
+    }
+
+    public async Task<CasdoorResponse> DoPostAsync(string action,
+        Dictionary<string, string>? queryMap, byte[] postBytes, bool isFile)
+    {
+        string url = GetUrl(action, queryMap ?? new Dictionary<string, string>());
+        using var handler = new HttpClientHandler
+        {
+            Credentials = new NetworkCredential(_options.ClientId, _options.ClientSecret)
+        };
+        using HttpClient httpClient = new(handler);
+        Task<HttpResponseMessage> postTask;
+
+        if (isFile) // postFileAsync
+        {
+            using var formData = new MultipartFormDataContent();
+            formData.Add(new ByteArrayContent(postBytes), "file", "file");
+            postTask = httpClient.PostAsync(url, formData);
+        }
+        else // postStringAsync
+        {
+            StringContent postContent = new(
+                postBytes.ToString() ??
+                throw new CasdoorException("postBytes.ToString() returns null"),
+                Encoding.UTF8,
+                "text/plain");
+            postTask = httpClient.PostAsync(url, postContent);
+        }
+
+        var resp = await postTask;
+        using MemoryStream ms = new(Encoding.UTF8.GetBytes(resp.Content.ToString()
```

Comment:
The memory stream usage here is wrong, we can direct;y call read as stream here.
https://docs.microsoft.com/zh-cn/dotnet/api/system.net.http.httpcontent.readasstreamasync?view=net-6.0#system-net-http-httpcontent-readasstreamasync(system-threading-cancellationtoken)